### ISSUE TITLE
Device info doesnt return device model 

### ISSUE THREAD

**Issue by nikitabalabanovdev** [matinzd/react-native-health-connect#163]
@matinzd 
**### Device metadata always returns null values in Health Connect library**

### Description: 

When fetching activity data, the metadata?.device object consistently returns {"manufacturer": null, "model": null, "type": 0}. This behavior occurs regardless of whether the code is running on an emulator or a real device. The expectation is that the manufacturer and model fields should return the device information for real devices and appropriate metadata for emulators.

### Steps to reproduce the behavior:

1. Use the Health Connect library to fetch activity data (e.g., steps).
2. Check the response in the metadata?.device field.
3. Observe that manufacturer, model, and type are always null or 0.

### Expected behavior:

The metadata?.device should return the actual device details

### Minimal Reproducible

Here’s a code snippet demonstrating the issue:
```
const { records: stepCountRecords } = await readRecords('Steps', {
  timeRangeFilter: {
    operator: 'between',
    startTime: startDate,
    endTime: endDate,
  },
});

if (stepCountRecords && stepCountRecords.length > 0) {
  const stepFormatted = {
    anchor: endDate,
    fitnessData: stepCountRecords
      .filter(
        record => record.count > 0 && !!record.startTime && !!record.endTime
      )
      .map(({ count, startTime, endTime, metadata }) => {
        const recordingMethod = metadata?.recordingMethod ?? 0;
        const device = metadata?.device?.model ?? emptyValue;
        const dataOrigin = metadata?.dataOrigin ?? emptyValue;
        return metadata?.device;
      }),
  };
}
```
### Environment:

Health Connect Version: ^3.1.0
React Native Version: 0.72.7
New architecture enabled: No
Using Expo: No
Android API Level: API 34 (Android 14)

**Comment by matinzd** [comment]
Hey!

Can you debug the app with Android Studio and add a breakpoint here? Then check what's in the `record.metadata`.

https://github.com/matinzd/react-native-health-connect/blob/d7faf11dea310e231aa0dabd5f645dc983d103a9/android/src/main/java/dev/matinzd/healthconnect/records/ReactStepsRecord.kt#L26-L33

If the recording method is the same as you see in the javascript side, there is nothing I can help you with.

**Comment by nikitabalabanovdev** [comment]
Yes  recording method is the same. Does it mean that this problem is not in the library, but in Google Fit or Health Connect itself ?

**Comment by matinzd** [comment]
How about other information like device data? Yes, then the problem is not the problem with react native lib. You need to open an issue in Google Issue Tracker.

**Comment by nikitabalabanovdev** [comment]
@matinzd  All the data is same. I get it, thanks a lot for you responses. I will open an issue in Google Issue Tracker

**Comment by nikitabalabanovdev** [comment]
@matinzd I am super sorry for bothering you, but have you seen latest changes of 
[feature request](https://github.com/matinzd/react-native-health-connect/issues/159)
It seems that it is more issue report than feature request

**Comment by matinzd** [comment]
@nikitabalabanovdev What do you mean?
